### DISCUSSION TITLE
Mu-Editor & PyQt6

### DISCUSSION THREAD

**Discussion by pdalet** [mu-editor/mu#2508] [General]
Hello,
First, Thanks a lot for your simple and sublime editor.

I use "Mu Editor" with Python 3.12 with PyQt6-Qt6 6.6.3 (ubuntu 22.04 and win10). It works fine.

For information, I found a little bug with Ctrl+Shift+C (context menu with PyQt6)

/usr/local/lib/python3.12/site-packages/mu/interface/interface/panes.py
line 1194

        self.cp=QShortcut(copy_keys, self)
        self.cp.activated.connect(self.copy)
        action_copy=menu.addAction("Copy  Ctrl+Shift+C")
        action_copy.triggered.connect(self.copy)
        #menu.addAction("Copy", self.copy, copy_keys) ERROR
        
        self.pe=QShortcut(paste_keys, self)
        self.pe.activated.connect(self.paste)
        action_paste=menu.addAction("Paste  Ctrl+Shift+V")
        action_paste.triggered.connect(self.paste)
        #menu.addAction("Paste", self.paste, paste_keys)  ERROR

Best regards
Philippe DALET

**Comment by carlosperate** [comment]
Thanks for letting us know @pdalet!
How did you use PyQt5 with Mu, was it from this PR?
https://github.com/mu-editor/mu/pull/2279

**Comment by pdalet** [comment]
Hello

I'm not using PyQt5, but PyQt6 with python 3.13 on Window10, Ubuntu 22.04 ans 24.04. 

It works fine.

I've downloading the PyQt6 branch from Github and modified some files.



Philippe DALET
use this mail
***@***.***

**Reply by carlosperate** [reply]
That's great! Are you able to submit a PR against the pyqt6 branch with the changes you made to get it to run? That would help us when doing the final update to PyQt6.

**Comment by pdalet** [comment]
Le 12/24/24 02:27 PM, Carlos Pereira Atencio  ***@***.***> a écrit : 
> 
> 
> 
> That's great! Are you able to submit a PR against the pyqt6 branch with the changes you made to get it to run? That would help us when doing the final update to PyQt6.
> 
> —
> Reply to this email directly, view it on GitHub(https://github.com/mu-editor/mu/discussions/2508#discussioncomment-11658252 ), or unsubscribe(https://github.com/notifications/unsubscribe-auth/AJEFLJKQH654HW7WIPJQXOL2HFOKDAVCNFSM6AAAAABTYMKIIWVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTCNRVHAZDKMQ ).
> You are receiving this because you were mentioned.
> 
> 



Hello,

Sorry for my late reply.

This mail adress is obsolete. ***@***.*** is my favorite adress.

I join a file with modificattions for qt6. I use Pyqt6 6.8.1 + python 3.13.2

Ph.DALET

debugger\runner.py
line 456 replace
        Respond to a closed socket (not a user command, but needs handling).
---------------------------------------------------------------------------------------------------------------------------------------------------
interface\editor.py
line 445 replace
                self.annotate(line, text, 0)
                #self.annotate(line, text, self.annotationDisplay())
---------------------------------------------------------------------------------------------------------------------------------------------------
interface\pane.py
line 51 add
    QShortcut,

line 1194/1195 replace
        menu.addAction("Copy", self.copy, copy_keys)
        menu.addAction("Paste", self.paste, paste_keys)
by
        self.cp=QShortcut(copy_keys, self)
        self.cp.activated.connect(self.copy)
        action_copy=menu.addAction("Copy  Ctrl+Shift+C")
        action_copy.triggered.connect(self.copy)
        #menu.addAction("Copy", self.copy, copy_keys)
        
        self.pe=QShortcut(paste_keys, self)
        self.pe.activated.connect(self.paste)
        action_paste=menu.addAction("Paste  Ctrl+Shift+V")
        action_paste.triggered.connect(self.paste)
        #menu.addAction("Paste", self.paste, paste_keys)
---------------------------------------------------------------------------------------------------------------------------------------------------
modes\base.py
line 209 replace
    code_template = "# -*- coding: utf-8 -*-"
 ---------------------------------------------------------------------------------------------------------------------------------------------------
__init.py__
line 9 replace
__version__ = "1.2.2"
----------------------------------------------------------------------------------------------------------------------------------------------------
